examples/jenkins/pipeline/jenkinstemplate.json

```diff
@@ -37,6 +37,10 @@
       "apiVersion": "v1",
       "metadata": {
         "name": "${JENKINS_SERVICE_NAME}",
+        "annotations": {
+          "service.alpha.openshift.io/dependencies": "[{\"name\": \"jenkins-jnlp\", \"namespace\": \"\", \"kind\": \"Service\"}]",
```

Comment:
> Ugh, I'm really concerned about not using Kinds.  Most clients are kind
> focused when it comes to doing references.

I think that `Scale` clearly demonstrates the failing of building references based on an ambiguous attribute.  This code seems to be looking at a resource.  We could put a resource into the `kind` field so it serializes cleanly, but I worry about having an API that is already unable to express certain kinds of references.